File: codeforces/0907A_-_Masha_and_Bears/main.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long int ll;

bool climb(int a, int b) {
  return a <= b;
}

bool likes(int a, int b) {
  return climb(a, b) && 2 * a >= b;
}

int main() {
  int papa, mother, son, martha;
  cin >> papa >> mother >> son >> martha;
  for (int car3 = 1; car3 <= 300; car3++) {
    if (!likes(son, car3) || !likes(martha, car3))
      continue;
    for (int car2 = car3 + 1; car2 <= 300; car2++) {
      if (!likes(mother, car2) || !climb(martha, car2) || likes(martha, car2))
        continue;
      for (int car1 = car2 + 1; car1 <= 300; car1++) {
        if (!likes(papa, car1) || !climb(martha, car1) || likes(martha, car1))
          continue;
        cout << car1 << "\n" << car2 << "\n" << car3 << "\n";
        return 0;
      }
    }
  }
  cout << "-1\n";
  return 0;
}

File: codeforces/0907A_-_Masha_and_Bears/main.py
def main():
    def climb(a, b):
        return a <= b
    def like(a, b):
        return climb(a, b) and 2 * a >= b
    father, mother, son, martha = map(int, input().split(' '))
    for C3 in range(1, 301):
        if not like(son, C3) or not like(martha, C3):
            continue
        for C2 in range(C3 + 1, 301):
            if not like(mother, C2) or not climb(martha, C2) or like(martha, C2):
                continue
            for C1 in range(C2 + 1, 301):
                if not like(father, C1) or not climb(martha, C1) or like(martha, C1):
                    continue
                return '{}\n{}\n{}'.format(C1, C2, C3)
    return -1
print(main())
